### DISCUSSION TITLE
How to add horizontal lines to create a table?

### DISCUSSION THREAD

**Discussion by juanpe1922** [ImageMagick/ImageMagick#3450] [Help] [answered]
Hi, 

I have a very large table that looks like this
<img width="904" alt="table1" src="https://user-images.githubusercontent.com/42684446/112661846-5a06ce00-8e2d-11eb-8fbe-6d8a5f52e588.png">
Is there a way I could put horizontal lines between the rows, so that I get 'cells'? I tried appending a black line every X pixels, but the spacing of the table is not consistent, so I end up crossing a bunch of the numbers in the rows. I think my ideal solution would be something that 'scans' the page from top to bottom, and adds a horizontal line whenever it finds, say 3 white pixels in a row.  But I don't know how to do this.

Thank you in advance!

**Comment by snibgo** [comment]
It would need to be in a shell script, but you don't say what shell you use (eg bash or Windows). Also, what version of IM?

**Reply by juanpe1922** [reply]
Thanks, sorry I didn't include that information. I use IM version: 7.0.11-2 Q16 x64 2021-02-28 on Windows 10, but I could also use bash if it's easier to do it there.

**Comment by snibgo** [comment, marked as answer]
The easy solution, in a single command (Windows BAT syntax), is:
```
magick ^
  112661846-5a06ce00-8e2d-11eb-8fbe-6d8a5f52e588.png ^
  ( +clone ^
    -alpha off ^
    -set option:WW %%[fx:w] ^
    -scale "1x^!" ^
    -threshold 95%% ^
    -negate ^
    -scale "%%[WW]x^!" ^
  ) ^
  -compose Darken -composite ^
  out.png
```
<img width="904" alt="x" src="https://user-images.githubusercontent.com/33812368/112674093-6c006500-8e5d-11eb-866b-3b9a3ce9d02a.png">

However, this gives thick lines. I expect these can be thinned with morphology, but I don't have time right now to figure it out.

EDIT to add:
Thinning the lines to one pixel wide, and also removing the top and bottom lines:
```
magick ^
  112661846-5a06ce00-8e2d-11eb-8fbe-6d8a5f52e588.png ^
  ( +clone ^
    -alpha off ^
    -set option:WW %%[fx:w] ^
    -scale "1x^!" ^
    -threshold 95%% ^
    -morphology Thinning:-1 skeleton ^
    -negate ^
    -scale "%%[WW]x^!" ^
  ) ^
  -compose Darken -composite ^
  out.png
```
<img width="904" alt="x" src="https://user-images.githubusercontent.com/33812368/112687727-cefaf780-8e6f-11eb-8610-b8bc5c88ce8b.png">

**Reply by juanpe1922** [reply]
Super. This is exactly what I needed. Thank you very much for your help!
